syntax = "proto3";
package logprog.v1;
// START: service
service Logging {
  rpc Produce(ProduceRequest) returns (ProduceResponse) {}
  rpc Consume(ConsumeRequest) returns (ConsumeResponse) {}
  rpc ConsumeStream(ConsumeRequest) returns (stream ConsumeResponse) {}
  rpc ProduceStream(stream ProduceRequest) returns (stream ProduceResponse) {}
}
// END: service

// START: apis
message ProduceRequest  {
   Record record = 1;
}

message ProduceResponse  {
   uint64 offset = 1;
}

message ConsumeRequest {
   uint64 offset = 1;
}

message ConsumeResponse {
  Record record = 2;
}
// END: apis

message Record {
   bytes value = 1;
   uint64 offset = 2;
}